### PR TITLE
packages/gcp_vertexai: Add support for regions and zone

### PR DESCRIPTION
**Pull request by ishleenk17**
## Proposed commit message

Add support for regions and zone in GCP Vertex AI Integration

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Related issues

- Closes  https://github.com/elastic/integrations/issues/12188

## Screenshots

Before applying Region Filter:
Screenshot 2024-12-26 at 3 59 08 PM

Region Filter in config:
Screenshot 2024-12-26 at 3 59 22 PM

After applying the Region Filter:
Screenshot 2024-12-26 at 3 58 55 PM

### CODE REVIEW

File: packages/gcp_vertexai/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.3.0"
+  changes:
+    - description: Add support of region/zone.
```
Comment (devamanv) [inline packages/gcp_vertexai/changelog.yml:4]:
```suggestion
    - description: Add support for regions and zone.
```

File: packages/gcp_vertexai/data_stream/metrics/agent/stream/stream.yml.hbs [hunk 2]
```diff
@@ -7,6 +7,15 @@ credentials_file_path: {{credentials_file}}
...
 credentials_json: '{{credentials_json}}'
 {{/if}}
+{{#if zone}}
+zone: {{zone}}
+{{/if}}
+{{#if regions}}
+regions:
+{{#each regions as |region i|}}
+- {{region}}
+{{/each}}
```
Comment (devamanv) [inline packages/gcp_vertexai/data_stream/metrics/agent/stream/stream.yml.hbs:17]:
```suggestion
{{#each regions as |region|}}
  - {{region}}
{{/each}}
```
Also block parameter `i` isn't used, so can be removed.

**Review by devamanv — APPROVED** [review]

File: packages/gcp_vertexai/data_stream/metrics/manifest.yml [hunk 3]
```diff
@@ -17,3 +17,18 @@ streams:
         multi: false
         required: false
         show_user: true
+      - name: zone
```
Comment (shmsr) [inline packages/gcp_vertexai/data_stream/metrics/manifest.yml:20]: no description for this?

File: packages/gcp_vertexai/data_stream/metrics/manifest.yml [hunk 4]
```diff
@@ -17,3 +17,18 @@ streams:
...
+        type: text
+        title: GCP Zone
+        multi: false
+        required: false
+        show_user: true
+        default: us-central1-a
+      - name: regions
+        type: text
+        title: GCP Regions
+        description: A list of GCP regions to pull data from
```
Comment (shmsr) [inline packages/gcp_vertexai/data_stream/metrics/manifest.yml:30]:
```suggestion
        description: List of GCP regions to fetch data from
```

**Review by shmsr — APPROVED** [review]
Couple of nitpicks; also +1 for Aman's comments. Rest looks good.

**Pull request merged**